### PR TITLE
VZ-4411 Update to TestFramework

### PR DESCRIPTION
**Pull request by bryantidd**
# Description

Please include a summary of the change and which issue is fixed.  If there are any dependencies, for example on a PR in another repository, please list those.

Fixes VZ-4411

# Checklist 

As the author of this PR, I have:

- [X] Checked that I included or updated copyright and license notices in all files that I altered
- [X] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: pkg/test/framework/ginkgo_wrapper.go [hunk 1]
```diff
@@ -5,15 +5,203 @@ package framework
...
+type TestFramework struct {
+	Pkg     string
+	Metrics *zap.SugaredLogger
+	Logs    *zap.SugaredLogger
+}
+
+func NewTestFramework(pkg string) *TestFramework {
+	t := new(TestFramework)
+	t.Pkg = pkg
+	t.Metrics, _ = metrics.NewLogger(pkg, metrics.MetricsIndex)
```
Comment (markxnelson) [inline pkg/test/framework/ginkgo_wrapper.go:27]: can we use reflection to get the pkg so they don't have to specify it? can be in a future iteration

File: pkg/test/framework/metrics/ginkgo_metrics.go [hunk 2]
```diff
@@ -86,7 +71,8 @@ func NewMetricsLogger(pkg string) (*zap.SugaredLogger, error) {
 		},
 	}
 
-	outputPaths, err := configureOutputs()
+	outputPaths, err := configureOutputs(ind)
+	logger.Info("boom")
```
Comment (markxnelson) [inline pkg/test/framework/metrics/ginkgo_metrics.go:75]: is this intended?

File: tests/e2e/examples/helidon/helidon_example_test.go [hunk 3]
```diff
@@ -46,16 +47,16 @@ var _ = framework.VzBeforeSuite(func() {
 		Eventually(func() error {
 			return pkg.CreateOrUpdateResourceFromFile("examples/hello-helidon/hello-helidon-app.yaml")
 		}, shortWaitTimeout, shortPollingInterval).ShouldNot(HaveOccurred(), "Failed to create hello-helidon application resource")
-		metrics.Emit(metricsLogger.With("deployment_elapsed_time", time.Since(start).Milliseconds()))
+		metrics.Emit(t.Metrics.With("deployment_elapsed_time", time.Since(start).Milliseconds()))
```
Comment (markxnelson) [inline tests/e2e/examples/helidon/helidon_example_test.go:50]: is this an arbitrary metric for this test itself? not a framework one?

File: tests/e2e/examples/helidon/helidon_example_test.go [hunk 4]
```diff
@@ -89,13 +90,15 @@ const (
 	ingressServiceName = "istio-ingressgateway"
 )
 
-var _ = framework.VzDescribe("Verify Hello Helidon OAM App.", func() {
+var _ = t.AfterEach(func() {})
```
Comment (markxnelson) [inline tests/e2e/examples/helidon/helidon_example_test.go:93]: can we make it so this is not needed?

File: pkg/test/framework/metrics/ginkgo_metrics.go (hunk 2 above)
Comment (bryantidd) [reply pkg/test/framework/metrics/ginkgo_metrics.go:75]: I think that was leftover from testing

File: tests/e2e/examples/helidon/helidon_example_test.go (hunk 3 above)
Comment (bryantidd) [reply tests/e2e/examples/helidon/helidon_example_test.go:50]: yes

File: tests/e2e/examples/helidon/helidon_example_test.go (hunk 4 above)
Comment (bryantidd) [reply tests/e2e/examples/helidon/helidon_example_test.go:93]: I'll see

**Review by markxnelson — APPROVED** [review]

**Pull request merged**